### PR TITLE
Bugfix: allow folder root name

### LINKED ISSUE #393: File explorer is broken when creating a folder with the same name as the root folder
**Describe the bug**

File explorer is broken when creating a folder with the same name as the root folder. We can no longer create another file or folder.

**To Reproduce**

https://github.com/ditrit/leto-modelizer/assets/62500081/6042be98-b59a-4ef8-b8ad-ac53a75d02d6

**Expected behavior**

Be able to create the folder with the same name.

**Application :**

- leto version [1.2.0], commit hash 7276fca8f1a41a24ce925c5b6bf7c1bc34836282

### LINKED ISSUE #465: Missing file name in config file when using githubator-plugin
**Describe the bug**

After adding components, we don't see the file name in the path in the config file. If we return to the draw view and update the position (click on `rearrange` button in this example), then get back to text view, we see a new object with updated position and the full 'path' (with the file name) of the corresponding components.
This bug is probably linked to [this one](https://github.com/ditrit/leto-modelizer/issues/351)

**To Reproduce**

https://github.com/ditrit/leto-modelizer/assets/62500081/5ab1adc9-c41b-41a6-b021-8d6f6183f9d7

**Expected behavior**

Instead of this config file:

```js
{
".github/workflows": {
    "githubator-plugin": {
      "workflow_1": {
        "x": 30,
        "y": 30,
        "width": 175,
        "height": 340,
        "needsResizing": false,
        "needsPositioning": false,
        "manuallyResized": false
      },
      "job_1": {
        "x": 30,
        "y": 30,
        "width": 110,
        "height": 195,
        "needsResizing": false,
        "needsPositioning": false,
        "manuallyResized": false
      }
    }
  },
  ".github/workflows/new_workflow.yml": {
    "githubator-plugin": {
      "workflow_1": {
        "x": 12,
        "y": 12,
        "width": 157,
        "height": 322,
        "needsResizing": false,
        "needsPositioning": false,
        "manuallyResized": false
      },
      "job_1": {
        "x": 12,
        "y": 12,
        "width": 110,
        "height": 195,
        "needsResizing": false,
        "needsPositioning": false,
        "manuallyResized": false
      }
    }
  },
}
```

We expect to have a valid config file (no duplication and containing the **full path**) like so :

```js
{
  ".github/workflows/new_workflow.yml": {
    "githubator-plugin": {
      "workflow_1": {
        "x": 12,
        "y": 12,
        "width": 157,
        "height": 322,
        "needsResizing": false,
        "needsPositioning": false,
        "manuallyResized": false
      },
      "job_1": {
        "x": 12,
        "y": 12,
        "width": 110,
        "height": 195,
        "needsResizing": false,
        "needsPositioning": false,
        "manuallyResized": false
      }
    }
  },
}
```

**Application:**

- leto version 1.3.0
- leto-modelizer-plugin-core: 0.22.0
- githubator-plugin: 0.3.0

### LINKED ISSUE #474: Delete last component of model file, the file is deleted and model is no longer displayed in the list
**Describe the bug**

After deleting the last component from model file, the file is deleted and the model is no longer displayed in the list.

**To Reproduce**

https://github.com/ditrit/leto-modelizer/assets/62500081/a56a3b1b-c580-4139-a1f5-bace5be72144

**Expected behavior**

 After deleting the last component from model file, the file must be preserved with empty content.

**Application :**

- leto version 1.3.0
- githubator-plugin 0.3.0

### PR DESCRIPTION
**Pull request by justinehell**
Resolves #393
Resolves #465
Resolves #474

### CODE REVIEW

File: src/composables/FileExplorer.js [hunk 1]
```diff
@@ -54,9 +56,13 @@ export function createFile(fileInformation, parentFolder, name) {
  * @param {number} index - Index of the part to check in the path.
  */
 export function updateFileInformation(parentNode, fileStatus, index = 0) {
-  const splitPath = fileStatus.path.split('/');
+  const splitPath = fileStatus.path.split('/').slice(1);
   const item = parentNode.children.find((child) => child.label === splitPath[index]);
 
+  if (!item) {
```
Comment (k-y0u) [inline src/composables/FileExplorer.js:62]: I don't see a case where this if would be used.

**Review by k-y0u — CHANGES_REQUESTED** [review]

File: src/composables/FileExplorer.js [hunk 2]
```diff
@@ -54,9 +56,13 @@ export function createFile(fileInformation, parentFolder, name) {
  * @param {number} index - Index of the part to check in the path.
  */
 export function updateFileInformation(parentNode, fileStatus, index = 0) {
-  const splitPath = fileStatus.path.split('/');
+  const splitPath = fileStatus.path.split('/').slice(1);
```
Comment (Zorin95670) [inline src/composables/FileExplorer.js:59]: can you add a comment to explain why you have to slice 1

File: src/composables/Git.js [hunk 3]
```diff
@@ -189,6 +189,7 @@ export async function createBranchFrom(projectId, newBranchName, branchName) {
  * Add untracked, unstaged or modified files.
  * @param {string} projectId - Id of project.
  * @param {string} filepath - Path of the file to add.
+ * NOTE: Filepath must exclude the root folder.
```
Comment (Zorin95670) [inline src/composables/Git.js:192]: @param {string} filepath - Path of the file to add. Filepath must exclude the root folder.

Same  for other

File: cypress/e2e/Bugfix/Issue393.feature [hunk 4]
```diff
@@ -0,0 +1,41 @@
+Feature: Fix issue #393: File explorer is broken when creating a folder with the same name as the root folder
+
+  Scenario: File explorer can display a folder with the same name as the root folder
+    Given I clear cache
+    And   I set viewport size to '1920' px for width and '1080' px for height
+    And   I visit the '/'
+    And   I wait until the application is loaded
+    And   I set context field 'projectName' with 'projectName'
```
Comment (Zorin95670) [inline cypress/e2e/Bugfix/Issue393.feature:8]: remove this

Comment (Zorin95670) [reply cypress/e2e/Bugfix/Issue393.feature:8]: but use reals project name like LETO

File: changelog.md [hunk 5]
```diff
@@ -51,6 +51,10 @@ and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0
 * Fix bug about (Link|Reference)Input options update, see [this issue](https://github.com/ditrit/leto-modelizer/issues/390).
 * Fix bug about adding ellipsis on long file name in file explorer, see [this issue](https://github.com/ditrit/leto-modelizer/issues/383).
 * Fix Dockerfile by adding git in order to install plugins.
+* Fix bug on creating folder with same name as root folder, see [this issue](https://github.com/ditrit/leto-modelizer/issues/393).
+* Fix bug on missing file name in config file when using githubator-plugin, see [this issue](https://github.com/ditrit/leto-modelizer/issues/465).
+* Fix bug on deleting last component of model file, the file is deleted and model is no longer displayed in the list, see [this issue](https://github.com/ditrit/leto-modelizer/issues/474).
+
```
Comment (Zorin95670) [inline changelog.md:57]: delete this line

File: src/composables/FileExplorer.js (hunk 2 above)
Comment (justinehell) [reply src/composables/FileExplorer.js:61]: yes

File: src/composables/Git.js (hunk 3 above)
Comment (justinehell) [reply src/composables/Git.js]: done

File: changelog.md (hunk 5 above)
Comment (justinehell) [reply changelog.md:57]: done

File: src/composables/PluginManager.js [hunk 6]
```diff
@@ -221,6 +209,38 @@ export async function renderModel(projectId, modelPath, plugin) {
   ]).then(() => filesToUpdate);
 }
 
+/**
+ * Render the given model with the corresponding pugin.
```
[the remark this answers is not in the document]
Comment (justinehell) [reply src/composables/PluginManager.js]: done

File: cypress/e2e/Bugfix/Issue393.feature (hunk 4 above)
Comment (justinehell) [reply cypress/e2e/Bugfix/Issue393.feature]: done

**Review by Zorin95670 — APPROVED** [review]

**Pull request merged**